typed_sql/lib/src/dialect/postgres_dialect.dart

```diff
@@ -0,0 +1,521 @@
+      };
+}
+
+extension on QueryContext {
+  String parameter(Object? value) {
+    switch (value) {
+      case int _:
+        final index = param(value);
+        return '\$$index::BIGINT';
+      case double _:
+        final index = param(value);
+        return '\$$index::DOUBLE PRECISION';
+      case Uint8List _:
+        final index = param(value);
+        return '\$$index::BYTEA';
+      case DateTime _:
+        final index = param(value.toUtc());
+        return '\$$index::TIMESTAMP WITH TIME ZONE';
+    }
+    final index = param(value);
+    return '\$$index';
+  }
+}
+
+final _sqlSafe = RegExp(r'^[a-zA-Z][a-zA-Z0-9_]*$');
+
+extension on CompositeQueryClause {
+  String get operator => switch (this) {
+        UnionClause() => 'UNION',
+        UnionAllClause() => 'UNION ALL',
+        IntersectClause() => 'INTERSECT',
+        ExceptClause() => 'EXCEPT',
+      };
+}
+
+extension on BinaryOperationExpression {
+  String get operator => switch (this) {
+        // TODO: Consider if some sort of type hierachy could possibly make this
+        //       even smarter.
+        ExpressionBoolAnd() => 'AND',
+        ExpressionBoolOr() => 'OR',
+        ExpressionStringEquals() => 'IS NOT DISTINCT FROM',
+        ExpressionStringLessThan() => '<',
+        ExpressionStringLessThanOrEqual() => '<=',
+        ExpressionStringGreaterThan() => '>',
+        ExpressionStringGreaterThanOrEqual() => '>=',
+        ExpressionNumEquals<num>() => 'IS NOT DISTINCT FROM',
+        ExpressionNumAdd<num>() => '+',
+        ExpressionNumSubtract<num>() => '-',
+        ExpressionNumMultiply<num>() => '*',
+        ExpressionNumDivide<num>() => '/',
+        ExpressionNumLessThan<num>() => '<',
+        ExpressionNumLessThanOrEqual<num>() => '<=',
+        ExpressionNumGreaterThan<num>() => '>',
+        ExpressionNumGreaterThanOrEqual<num>() => '>=',
+        ExpressionDateTimeEquals() => 'IS NOT DISTINCT FROM',
+        ExpressionDateTimeLessThan() => '<',
+        ExpressionDateTimeLessThanOrEqual() => '<=',
+        ExpressionDateTimeGreaterThan() => '>',
+        ExpressionDateTimeGreaterThanOrEqual() => '>=',
+      };
+}
+
+extension on ColumnType {
+  String get sqlType => switch (this) {
+        ColumnType<Uint8List> _ => 'BYTEA',
+        ColumnType<bool> _ => 'BOOLEAN',
+        ColumnType<DateTime> _ => 'TIMESTAMP WITH TIME ZONE',
+        ColumnType<int> _ => 'BIGINT',
+        ColumnType<double> _ => 'DOUBLE PRECISION',
+        ColumnType<String> _ => 'TEXT',
+        ColumnType<Null> _ => throw UnsupportedError(
+            'Null type cannot be used as column type',
+          ),
+      };
+}
+
+extension on JoinType {
+  String get kind => switch (this) {
+        JoinType.inner => 'INNER',
+        JoinType.left => 'LEFT',
+        JoinType.right => 'RIGHT',
+      };
+}
+
+/// Reserved keywords in sqlite
+///
+/// See: https://sqlite.org/lang_keywords.html
+final _sqliteReservedKeywords = <String>[];
```

Comment:
Why is this defined in postgres_dialect.dart?

Reply:
Should be renamed and updated.